src/ray/common/task/task_spec.cc

```diff
@@ -187,8 +187,11 @@ ObjectID TaskSpecification::ActorDummyObject() const {
 }
 
 bool TaskSpecification::IsDirectCall() const {
-  RAY_CHECK(IsActorCreationTask());
-  return message_->actor_creation_task_spec().is_direct_call();
+  if (IsActorCreationTask()) {
+    return message_->actor_creation_task_spec().is_direct_call();
```

Comment:
could we also set the `is_direct_call` flag for direct actor tasks to avoid potential confusion?

Reply:
Hmm perhaps. It's a bit confusing at the moment because perhaps you could create an non-direct call actor with a direct call.

Long term the flag will be removed.